plugins/win-capture/game-capture.c

```diff
@@ -592,21 +601,109 @@ static void load_placeholder_image(struct game_capture *gc)
+	if (!dstr_is_empty(&gc->placeholder_image_path)) {
+		gs_image_file2_init(&gc->placeholder_image, gc->placeholder_image_path.array);
 
 		obs_enter_graphics();
-		gs_image_file2_init_texture(&gc->if2);
+		gs_image_file2_init_texture(&gc->placeholder_image);
 		obs_leave_graphics();
 	}
+	
+	TCHAR* translated_string = NULL;
+	size_t len = os_utf8_to_wcs(gc->placeholder_text.array, 0, NULL, 0);
+	if (len)
+	{
+		translated_string = malloc( (len+1)*2);
+		os_utf8_to_wcs(gc->placeholder_text.array, 0, &translated_string[0], len + 1);
+	} else {
+		return;
+	}
+
+	const int bytes_per_pixel = 4;
+	const int fraction_of_image_for_text = 8;
+	const int text_fitting_step = 5;
+	const int ALPHA_COMPONENT = 3;
+	struct obs_video_info ovi;	
+	obs_get_video_info(&ovi);
+
+	gc->placeholder_text_height = ovi.base_height/fraction_of_image_for_text;
+	gc->placeholder_text_width = ovi.base_width;
+
+	BITMAPINFOHEADER bmphdr = { 0 };
+	bmphdr.biSize = sizeof(BITMAPINFOHEADER);
+	bmphdr.biWidth = gc->placeholder_text_width;
+	bmphdr.biHeight = -gc->placeholder_text_height;
+	bmphdr.biPlanes = 1;
+	bmphdr.biBitCount = 32;
+	bmphdr.biSizeImage = gc->placeholder_text_height * gc->placeholder_text_width * bytes_per_pixel;
+	
+	HDC wdc = GetDC(NULL);
+	if (wdc) {
+		HDC memDC = CreateCompatibleDC ( wdc );
+
+		uint8_t * text_bitmap_buffer = bmalloc(gc->placeholder_text_height*gc->placeholder_text_width*bytes_per_pixel);	
+		HBITMAP text_bitmap = CreateDIBSection(NULL, (PBITMAPINFO)&bmphdr, DIB_RGB_COLORS, &text_bitmap_buffer, NULL, 0);
+		if (text_bitmap) {
+			SelectObject(memDC, text_bitmap);
+			SetTextColor(memDC, 0x00FFFFFF);
+			SetBkColor(memDC, 0x00000000);
+
+			int string_width = 0;
+
+			HFONT   font;
+			LOGFONT LogFont = {0};
+			LogFont.lfStrikeOut = 0;
+			LogFont.lfUnderline = 0;
+			LogFont.lfEscapement = 0;
+			LogFont.lfQuality = 0x05;
+			LogFont.lfItalic = 0;
+			LogFont.lfHeight = gc->placeholder_text_height;
+
+			while (true) {
+				font = CreateFontIndirect(&LogFont);
+				SelectObject(memDC, font);
+				
+				GetTextExtentPoint32(memDC, translated_string, len, &string_width);
+				if (string_width < gc->placeholder_text_width) 
+					break;
+				
+				DeleteObject(font);
+				LogFont.lfHeight = LogFont.lfHeight - text_fitting_step;
+			}
+
+			TextOut(memDC, (gc->placeholder_text_width-string_width)/2, (gc->placeholder_text_height-LogFont.lfHeight)/2, translated_string, len); 
+
+			for( int i = 0; i <gc->placeholder_text_height*gc->placeholder_text_width; i++)	{
+				int pixel_offset = i*bytes_per_pixel;
+				int text_pixel_color_components_average = (text_bitmap_buffer[pixel_offset+0] + text_bitmap_buffer[pixel_offset+1] + text_bitmap_buffer[pixel_offset+2])/3;
+				text_bitmap_buffer[pixel_offset + ALPHA_COMPONENT] = text_pixel_color_components_average;
+			}
+
+			obs_enter_graphics();
+			gc->placeholder_text_texture = gs_texture_create(gc->placeholder_text_width, gc->placeholder_text_height, GS_BGRA, 1, &text_bitmap_buffer, GS_DYNAMIC);
```

Comment:
Styling issues, from line 674 to 683 -> very hard to read, please add line return to ease the reading.